pkg/services/opsgenie/opsgenie_json.go

```diff
@@ -0,0 +1,75 @@
+package opsgenie
+
+import (
+	"encoding/json"
+	"github.com/containrrr/shoutrrr/pkg/types"
+)
+
+// JSON represents the  payload being sent to the OpsGenie API
+//
+// See: https://docs.opsgenie.com/docs/alert-api#create-alert
+//
+// Some fields contain complex values like arrays and objects.
+// Because `params` are strings only we cannot pass in slices
+// or maps. Instead we "preserve" the JSON in those fields. That
+// way we can pass in complex types as JSON like so:
+//
+//	service.Send("An example alert message", &types.Params{
+//		"alias":       "Life is too short for no alias",
+//		"description": "Every alert needs a description",
+//		"responders":  `[{"id":"4513b7ea-3b91-438f-b7e4-e3e54af9147c","type":"team"},{"name":"NOC","type":"team"}]`,
+//		"visibleTo":   `[{"id":"4513b7ea-3b91-438f-b7e4-e3e54af9147c","type":"team"},{"name":"rocket_team","type":"team"}]`,
+//		"details":     `{"key1": "value1", "key2": "value2"}`,
+//	})
+type JSON struct {
```

Comment:
Using the arbitrary type name `JSON` for this struct is a bit confusing. Perhaps something like `AlertPayload` or `AlertBody`?